Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -214,8 +177,8 @@ CGMutablePathRef CGPathCreateMutableCopy(CGPathRef path) {
     auto ret = __CGPath::alloc(nil);
     ret->_max = path->_max;
     ret->_count = path->_count;
-    ret->_components = (pathComponent*)IwRealloc(ret->_components, ret->_max * sizeof(pathComponent));
-    memcpy(ret->_components, path->_components, path->_count * sizeof(pathComponent));
+    ret->_components = (CGPathElement*)IwRealloc(ret->_components, ret->_max * sizeof(CGPathElement));
+    memcpy(ret->_components, path->_components, path->_count * sizeof(CGPathElement));
```

Comment:
This copy is shallow.  the _components->points is now shared between the two instances of CGPaths and will result in double free.